internal/provider/provisioner_key_resource_test.go

```diff
@@ -0,0 +1,96 @@
+package provider
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"text/template"
+
+	"github.com/coder/terraform-provider-coderd/integration"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAccProvisionerKeyResource(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Acceptance tests are disabled.")
+	}
+	ctx := context.Background()
+	client := integration.StartCoder(ctx, t, "license_acc", true)
+	orgs, err := client.Organizations(ctx)
+	require.NoError(t, err)
+	firstOrg := orgs[0].ID
+
+	license := os.Getenv("CODER_ENTERPRISE_LICENSE")
+	if license == "" {
+		t.Skip("No license found for license resource tests, skipping")
+	}
+
+	cfg1 := testAccProvisionerKeyResourceConfig{
+		URL:   client.URL.String(),
+		Token: client.SessionToken(),
+
+		OrganizationID: firstOrg,
+		Name:           "example-provisioner-key",
+	}
+
+	cfg2 := cfg1
+	cfg2.Name = "different-provisioner-key"
+	cfg2.Tags = map[string]string{
+		"wibble": "wobble",
+	}
+
+	resource.Test(t, resource.TestCase{
+		IsUnitTest:               true,
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: cfg1.String(t),
+			},
+			{
+				Config: cfg2.String(t),
+			},
```

Comment:
One test step modifying each of the attributes is probably good enough. You could also check the provisioner key attribute is populated with something that looks vaguely like a provisioner key.